localization/deviation_estimation_tools/deviation_evaluator/scripts/bag_load_utils.py

```diff
@@ -0,0 +1,336 @@
+        self.topic_type = {name_of: type_of for id_of, name_of, type_of in topics_data}
+        self.topic_id = {name_of: id_of for id_of, name_of, type_of in topics_data}
+        self.topic_msg_message = {
+            name_of: get_message(type_of) for id_of, name_of, type_of in topics_data
+        }
+
+    def __del__(self):
+        self.conn.close()
+
+    def get_messages(self, topic_name):
+        topic_id = self.topic_id[topic_name]
+        rows = self.cursor.execute(
+            "SELECT timestamp, data FROM messages WHERE topic_id = {}".format(topic_id)
+        ).fetchall()
+        return [
+            (timestamp, deserialize_message(data, self.topic_msg_message[topic_name]))
+            for timestamp, data in rows
+        ]
+
+
+class EKFBagFileParser(BagFileParser):
+    def __init__(self, bag_file):
+        super().__init__(bag_file)
+
+    def get_messages(self, topic_name):
+        topic_id = self.topic_id[topic_name]
+        rows = self.cursor.execute(
+            "SELECT timestamp, data FROM messages WHERE topic_id = {}".format(topic_id)
+        ).fetchall()
+        sample = deserialize_message(rows[0][1], self.topic_msg_message[topic_name])
+        if isinstance(sample, PoseWithCovarianceStamped):
+            return self.parse_pose_with_covariance_stamped(rows, topic_name)
+        elif isinstance(sample, TwistWithCovarianceStamped):
+            return self.parse_twist_with_covariance_stamped(rows, topic_name)
+        elif isinstance(sample, Odometry):
+            return self.parse_odometry(rows, topic_name)
+        else:
+            print(topic_name + " not found.")
+            return []
+
+    def parse_pose_with_covariance_stamped(self, rows, topic_name):
+        pose_list, pose_cov_list = [], []
+        for timestamp, data in rows:
+            msg = deserialize_message(data, self.topic_msg_message[topic_name])
+            R = Rotation.from_quat(
+                [
+                    msg.pose.pose.orientation.x,
+                    msg.pose.pose.orientation.y,
+                    msg.pose.pose.orientation.z,
+                    msg.pose.pose.orientation.w,
+                ]
+            )
+            _, _, yaw = R.as_euler("xyz", degrees=False)
+            pose_list.append(
+                [
+                    msg.header.stamp.sec + msg.header.stamp.nanosec * 1e-9,
+                    msg.pose.pose.position.x,
+                    msg.pose.pose.position.y,
+                    msg.pose.pose.position.z,
+                    yaw,
+                ]
+            )
+            pose_cov_list.append(msg.pose.covariance.reshape(6, 6)[(0, 1, 5), :][:, (0, 1, 5)])
+        sorted_idxs = np.argsort(np.array(pose_list)[:, 0])
+        pose_list = np.array(pose_list)[sorted_idxs]
+        pose_cov_list = np.array(pose_cov_list)[sorted_idxs]
+        pose_list = np.array(pose_list)
+        pose_cov_list = np.array(pose_cov_list)
```

Comment:
unnecessary lines

Reply:
Let me resolve this in the following PRs